src/client.rs

```diff
@@ -61,32 +75,44 @@ impl Client {
     async fn get<T: DeserializeOwned>(&self, path: &str) -> Result<T> {
         let mut retried = 0;
         let mut ms_between_retries = 10000;
+        let mut already_used_endpoints: Vec<String> = Vec::new();
+
         loop {
-            match self
-                .client
-                .get(&format!("{}/{}", self.next_endpoint(), path))
-                .send()
-                .await
-            {
+            let endpoint = self.next_endpoint(&already_used_endpoints);
+            let url = format!("{}/{}", endpoint, path);
+
+            match self.client.get(&url).send().await {
                 Ok(r) => match r.status() {
                     StatusCode::OK => match r.json::<T>().await {
                         Ok(r) => return Ok(r),
                         Err(e) => return Err(e.into()),
                     },
                     _ => {
+                        // If there's still endpoints not used yet, retry now with different endpoint
+                        if self.endpoints.len() != already_used_endpoints.len() {
+                            already_used_endpoints.push(endpoint.clone());
+                            continue;
+                        }
+
+                        // If all endpoints are used, wait then retry with random endpoint
```

Comment:
This comment is misleading as if `retry` is 3 but your have 5 endpoints, then it will not have exhausted all tries.